snuba_sdk/conditions.py

```diff
@@ -0,0 +1,55 @@
+from dataclasses import dataclass
+from enum import Enum
+from typing import Union
+
+from snuba_sdk import Expression
+from snuba_sdk.expressions import (
+    Column,
+    Function,
+    InvalidExpression,
+    Scalar,
+    ScalarType,
+    _stringify_scalar,
+)
+
+
+class Op(Enum):
+    GT = ">"
+    LT = "<"
+    GTE = ">="
+    LTE = "<="
+    EQ = "="
+    NEQ = "!="
+    IN = "IN"
+    NOT_IN = "NOT IN"
+    LIKE = "LIKE"
+    NOT_LIKE = "NOT LIKE"
+
+
+@dataclass(frozen=True)
+class Condition(Expression):
+    lhs: Union[Column, Function]
+    op: Op
+    rhs: Union[Column, Function, ScalarType]
+
+    def validate(self) -> None:
+        if not isinstance(self.lhs, (Column, Function)):
+            raise InvalidExpression(
+                f"invalid condition: LHS of a condition must be a Column or Function, not {type(self.lhs)}"
+            )
+        if not isinstance(self.rhs, (Column, Function, *Scalar)):
+            raise InvalidExpression(
+                f"invalid condition: RHS of a condition must be a Column, Function or Scalar not {type(self.rhs)}"
+            )
+        if not isinstance(self.op, Op):
+            raise InvalidExpression(
+                "invalid condition: operator of a condition must be an Op"
+            )
```

Comment:
Are you validating types since sentry does not use mypy ?

Reply:
Essentially, although I can't guarantee Sentry will be the only user of this package, so I think it's smart to do type checking on all the external interfaces.